backend/test_observer/data_access/repository.py

```diff
@@ -130,6 +139,35 @@ def get_artefacts_by_family(
     return query.all()
 
 
+def get_historic_test_executions(
+    session: Session,
+    test_execution: TestExecution,
+) -> list[TestExecution]:
+    current_artefact = test_execution.artefact_build.artefact
+    return (
+        session.query(TestExecution)
+        .join(TestExecution.artefact_build)
+        .join(ArtefactBuild.artefact)
+        .options(
+            contains_eager(TestExecution.artefact_build).contains_eager(
+                ArtefactBuild.artefact
+            )
+        )
+        .filter(
+            Artefact.name == current_artefact.name,
+            Artefact.store == current_artefact.store,
+            Artefact.repo == current_artefact.repo,
+            Artefact.series == current_artefact.series,
+            TestExecution.environment_id == test_execution.environment_id,
+            TestExecution.id < test_execution.id,
+        )
```

Comment:
perhaps you meant to use `Artefact.track == current_artefact.track` instead of `Artefact.store == current_artefact.store`

Reply:
Thanks for catching this, I forgot track is what we use to identify the artefact.